source/internal/_assertPromise.js

```diff
@@ -0,0 +1,8 @@
+import _isFunction from './_isFunction';
+import _toString from './_toString';
+
+export default function _assertPromise(name, p) {
+  if (p == null || !_isFunction(p.then)) {
+    throw new TypeError('`' + name + '` expected a Promise, received ' + _toString(p, []));
```

Comment:
Considering that these helpers are generic enough to work with then-ables, it might make sense to reflect that in the code and documentation.

Reply:
Perhaps, but at this point, that should be a separate PR.